lib/sinon/collection.js

```diff
@@ -46,7 +46,16 @@ var collection = {
     },
 
     resetHistory: function resetHistory() {
-        each(this, "resetHistory");
+        getFakes(this).forEach(function (fake) {
+            if (typeof fake.resetHistory === "function") {
+                fake.resetHistory();
+                return;
```

Comment:
> In my opinion, that requires a lot more effort to understand.

I would get the above code fixed in code review. It should be an `else-if`. I agree that the way it is written is more complex than an `else-if` although logically congruent.

> Using return as a control flow statement helps me keep indentation low, which I think we all agree makes code easier to read.

> I believe that using return to reduce indentation is a good practice.

We agree there, but an `else-if` is the same amount of indentation as a plain `if`: 

```js
if (foo) {
  // one indentation
  return;
}

if (bar) {
  // one indentation
}

// VS

if (foo) {
  // one indentation
} else if {
  // one indentation
}
```

You're not saving anything whatsoever in this case. You're actually typing a few extra characters if you're concerned about keystroke counts at all.

> try this search

We use empty returns all the time. That is not the/a problem. You're not getting the full story though.

```shell
ag --ignore=docs/ -Q "return;" -A 5 -B 3
```

I only found one instance of `if-return-if` usage. PS. I had been putting off installing/using `ag` until just now.

#### RE: consistency

Apparently every fake has `resetHistory` or `reset` so we should remove the second if and make it an `if-else` [like in spy](https://github.com/sinonjs/sinon/blob/36c6cdcec83dbeae7931c1949c1b1e79bf59e77b/lib/sinon/spy.js#L134-L142):

```js
        if (this.fakes) {
            this.fakes.forEach(function (fake) {
                if (fake.resetHistory) {
                    fake.resetHistory();
                } else {
                    fake.reset();
                }
            });
        }
```

Then this whole conversation goes away. :slightly_smiling_face: